scripts/forbid-todo.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+# Fail if there are any TODOs in the codebase
+
+# Find the string 'TODO' in all files tracked by git, excluding
+# this file
+TODOS_FOUND=$(git grep --color=always -nw TODO -- ':!scripts/forbid-todo.sh')
```

Comment:
Decided to just grep all files rather than try to find lint rules for different languages, to make sure we cover everything